src/types.rs

```diff
@@ -241,6 +231,45 @@ macro_rules! key_methods {
                 Ok(ret)
             }
         }
+
+        #[cfg(feature = "serde")]
+        impl serde::Serialize for $i {
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+            where
+                S: serde::Serializer,
+            {
+                use crate::alloc::string::ToString;
+                serializer.serialize_str(&self.to_string())
```

Comment:
Use `fmt` traits to eliminate alloc requirement.

```suggestion
serializer.serialize_str(&format!("{:#x}", &self))
```

Reply:
Thanks a lot! I was actually trying to do this but getting an error with `cargo check --verbose --target thumbv6m-none-eabi --no-default-features --features serde`. I will be trying to work that out.